src/Core/Services/CryptoService.cs

```diff
@@ -670,29 +692,39 @@ public async Task<EncByteArray> EncryptToBytesAsync(byte[] plainValue, Symmetric
 
         private async Task StoreAdditionalKeysAsync(UserKey userKey, string userId = null)
         {
-            // Refresh, set, or clear the pin key
+            // Set, refresh, or clear the pin key
             if (await _stateService.GetProtectedPinAsync(userId) != null)
             {
-                await UpdateUserKeyPinAsync(userKey, userId);
+                await UpdatePinKeyAsync(userKey, userId);
             }
             else
             {
                 await _stateService.SetPinKeyEncryptedUserKeyAsync(null, userId);
                 await _stateService.SetPinKeyEncryptedUserKeyEphemeralAsync(null, userId);
             }
 
-            // Refresh, set, or clear the auto key
+            // Set, refresh, or clear the auto unlock key
             if (await _stateService.GetVaultTimeoutAsync(userId) == null)
             {
-                await _stateService.SetUserKeyAutoUnlockAsync(userKey.KeyB64, userId);
+                await _stateService.SetUserKeyAutoUnlockAsync(userKey, userId);
             }
             else
             {
                 await _stateService.SetUserKeyAutoUnlockAsync(null, userId);
             }
+
+            // Set, refresh, or clear the biometric unlock key
+            if ((await _stateService.GetBiometricUnlockAsync(userId)).GetValueOrDefault())
```

Comment:
⛏️ There were some internal discussions about what to do on nullable bools and most comments pointed to using `is true`, so I'd change that here:
```suggestion
            if (await _stateService.GetBiometricUnlockAsync(userId) is true)
```
Also, if possible I'd change a bit the naming of the method to something like `GetCanBiometricUnlockAsync(...)` (`can`, `allow`, `has`, what makes most sense) to represent the intent clearer. It was a bit confusing to me what `get biometric unlock` means.

Reply:
Great callout! I wish I had time to refactor the biometric system a little more, but I'm worried about introducing more bugs this late. From what I've gathered:
- `BiometricUnlock` - Do we have biometrics enabled?
- `BiometricLocked` - Is our vault currently locked and requires biometrics to unlock?
- `BiometricUnlockKey` - What I've added, the biometric key